samcli/lib/telemetry/metric.py

```diff
@@ -158,6 +159,9 @@ def wrapped(*args, **kwargs):
             metric_specific_attributes["gitOrigin"] = get_git_remote_origin_url()
             metric_specific_attributes["projectName"] = get_project_name()
             metric_specific_attributes["initialCommit"] = get_initial_commit_hash()
+            # Event metrics
+            metric_specific_attributes["events"] = [e.to_json() for e in EventTracker.get_tracked_events()]
```

Comment:
Can we add some tests to validate:
1/ if some events are tracked, `metric_specific_attributes["events"]` contains the expected events; and
2/ if no event is tracked, `metric_specific_attributes["events"]` is an empty array

Reply:
Definitely! I wasn't sure whether tests such as those would be necessary since the attribute existence is covered by the existing metric test, but I'll implement them to be more specific and detailed to the events